sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcTestDataSet.java

```diff
@@ -78,32 +59,33 @@ public static PGSimpleDataSource getDataSource(IOTestPipelineOptions options)
 
   public static final String READ_TABLE_NAME = "BEAM_TEST_READ";
 
-  public static void createReadDataTable(DataSource dataSource) throws SQLException {
+  public static void createReadDataTableAndAddInitialData(DataSource dataSource)
+      throws SQLException {
     createDataTable(dataSource, READ_TABLE_NAME);
+    addInitialData(dataSource, READ_TABLE_NAME);
   }
 
-  public static String createWriteDataTable(DataSource dataSource) throws SQLException {
-    String tableName = "BEAMTEST" + org.joda.time.Instant.now().getMillis();
+  public static String createWriteDataTableAndAddInitialData(DataSource dataSource)
+      throws SQLException {
+    String tableName = getWriteTableName();
     createDataTable(dataSource, tableName);
+    addInitialData(dataSource, tableName);
     return tableName;
   }
 
-  private static void createDataTable(DataSource dataSource, String tableName) throws SQLException {
-    try (Connection connection = dataSource.getConnection()) {
-      // something like this will need to happen in tests on a newly created postgres server,
-      // but likely it will happen in perfkit, not here
-      // alternatively, we may have a pipelineoption indicating whether we want to
-      // re-use the database or create a new one
-      try (Statement statement = connection.createStatement()) {
-        statement.execute(
-            String.format("create table %s (id INT, name VARCHAR(500))", tableName));
-      }
+  public static String getWriteTableName() {
+    return "BEAMTEST" + org.joda.time.Instant.now().getMillis();
```

Comment:
done. I used underscores since it doesn't require escaping in either of the two DBs being used - derby/postgres.